src/overlays/actors/ovl_En_Baisen/z_en_baisen.c

```diff
@@ -21,30 +38,255 @@ const ActorInit En_Baisen_InitVars = {
+void func_80BE895C(EnBaisen* this, GlobalContext* globalCtx) {
+    if (this->unk2A4 != NULL) {
+        this->unk290 = 1;
+        this->unk2AC = 1;
+        func_800B86C8(this->unk2A4, globalCtx, this->unk2A4);
+    }
+    this->unk29C = 1;
+    if (this->paramCopy == 0) {
+        this->unk2A4 = this->heishiPointer;
+        this->actionFunc = func_80BE8AAC;
+    } else {
+        this->actionFunc = func_80BE89D8;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE87B0.asm")
+void func_80BE89D8(EnBaisen* this, GlobalContext* globalCtx) {
+    if (&this->actor == this->unk2A4) {
+        this->unk29E = this->actor.world.rot.y;
+        if (this->animIndex == 0) {
+            EnBaisen_ChangeAnimation(this, 1);
+        }
+    } else {
+        this->unk29E = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk2A4->world.pos);
+        if (this->animIndex != 0) {
+            EnBaisen_ChangeAnimation(this, 0);
+        }
+    }
+    if ((globalCtx->msgCtx.unk11F04 == 0x2AC6) || (globalCtx->msgCtx.unk11F04 == 0x2AC7) ||
+        (globalCtx->msgCtx.unk11F04 == 0x2AC8)) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+        this->unk29E = this->actor.yawTowardsPlayer;
+    }
+    if (this->unk2AC == 2) { // Note: This variable is only ever set to 1.
+        func_80BE87FC(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE87FC.asm")
+void func_80BE8AAC(EnBaisen* this, GlobalContext* globalCtx) {
+    if (this->textIdIndex & 1) {
+        this->unk29E = this->actor.world.rot.y;
+        if (this->animIndex == 0) {
+            EnBaisen_ChangeAnimation(this, 1);
+        }
+    } else {
+        if (this->unk2A4 != NULL) {
+            this->unk29E = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk2A4->world.pos);
+        }
+        if (this->animIndex != 0) {
+            EnBaisen_ChangeAnimation(this, 0);
+        }
+    }
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        func_801477B4(globalCtx);
+        this->textIdIndex++;
+        if (this->textIdIndex < 6) {
+            func_80151938(globalCtx, textIds[this->textIdIndex]);
+            if (!(this->textIdIndex & 1)) {
+                this->unk2A4 = this->heishiPointer;
+            } else {
+                this->unk2A4 = &this->actor;
+            }
+            func_800B86C8(this->unk2A4, globalCtx, this->unk2A4);
+        } else {
+            func_80BE87FC(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE887C.asm")
+void EnBaisen_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnBaisen* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE895C.asm")
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->unusedCounter != 0) {
+        this->unusedCounter--;
+    }
```

Comment:
The counter's value is never set to anything (and it's initialized to 0), so this piece of code doesn't actually ever do anything. Given the way that it checks its value against 0 and decrements it, to me it seems that it was intended to be some sort of counter, but since they never give this variable a value, I proclaimed it unused. Definitely open to change it though

Reply:
It could be possible that another actor sets it